lib/logstash/config/config_ast.rb

```diff
@@ -275,9 +277,17 @@ def compile
     end
   end
   class RValue < Node; end
-  class Value < RValue; end
+  class Value < RValue;
+    def text_value_utf8
+      Unicode.convert(text_value)
+    end
+  end
 
   module Unicode
+    def self.convert(text)
+      text.force_encoding(Encoding::UTF_8)
```

Comment:
note that this does not transcode but only flags the string as being UTF-8. not sure if this is your intended behaviour.